pom.xml

```diff
@@ -751,6 +746,184 @@
           </formats>
         </configuration>
       </plugin>
+
+      <plugin>
+        <groupId>org.jacoco</groupId>
+        <artifactId>jacoco-maven-plugin</artifactId>
+        <version>${maven.jacoco.plugin.version}</version>
+        <executions>
+          <execution>
+            <id>default-instrument</id>
+            <goals>
+              <goal>instrument</goal>
+            </goals>
+            <phase>test-compile</phase>
+          </execution>
+          <execution>
+            <id>default-restore-instrumented-classes</id>
+            <goals>
+              <goal>restore-instrumented-classes</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>default-report</id>
+            <phase>site</phase>
+            <goals>
+              <goal>report</goal>
+            </goals>
+            <configuration>
+              <excludes>
+                <!-- Parser related classes -->
+                <exclude>com/puppycrawl/tools/checkstyle/grammars/*.class</exclude>
+                <exclude>com/puppycrawl/tools/checkstyle/grammars/javadoc/*.class</exclude>
+                <exclude>com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser$JavadocParserErrorStrategy.class</exclude>
+                <!-- Swing related classes -->
+                <exclude>com/puppycrawl/tools/checkstyle/gui/*.class</exclude>
+                <!-- Deprecated classes -->
+                <exclude>com/puppycrawl/tools/checkstyle/checks/naming/AbstractTypeParameterNameCheck.class</exclude>
+                <!-- Seems to be a bug in JaCoCo. It complains about static init on a compiler generated class. -->
+                <exclude>com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationUseStyleCheck$1.class</exclude>
```

Comment:
Has the issue been reported to Jacoco?

Reply:
it might be hard to open good issue for this...

@pbludov , please create some issue with uncertain description, we will use link here.
After merge, steps to reproduce in jacoco issue will be simple: clone, remove exclude, run command.

In the same time we will have some tracking number here for future to recheck.
Comment should be like "till https:///////#XXXXX"

Reply:
This may be known issue, I'll take a look.

Reply:
No, there is no issues like that in the Jacoco tracker.
The problem is easy to reproduce, so I created a new task:
https://github.com/jacoco/jacoco/issues/660

Reply:
Please reference it in comment above suppression

Reply:
Thanks.

Reply:
@pbludov @romani jacoco/jacoco#660 is now closed and has a milestone of 0.8.2 .